### ISSUE TITLE
No parallel execution in PHP 8.1, and space character crashes parallel task

### ISSUE THREAD

**Issue by falxen** [krakjoe/parallel#238]
Hello, I am running PHP 8.1.7 ZTS on Nginx using PHP-FPM.

Parallel version is branch `dixyes-patch-1` of the fork [https://github.com/dixyes/parallel](https://github.com/dixyes/parallel) based on [#226](https://github.com/krakjoe/parallel/pull/226) because I couldn't get the main branch to compile with PHP 8.1.x. Unfortunately, I have to use PHP 8.1.x and can't use any 8.0.x version.

**Issue 1:**
```
<?php
use \parallel\Runtime;
$runtime = new Runtime();

$task = static function (): string {
   for ($i = 0; $i < 20; $i++):
      echo $i;
   endfor;
   return "easy";
};

$future = $runtime->run($task);

for ($i = 0; $i < 20; $i++):
   echo "*";
endfor;

echo "<br />";
printf("\nUsing \\parallel\\Runtime is %s\n", $future->value());
die();
```
I expect to get something like `*1*2*3*4` (etc) but instead I get only stars i.e.
```
********************
Using \parallel\Runtime is easy
```
so the runtime isn't executing though it's returning a value.

**Issue 2:**
If I change the numbers in that task to a space, then the webpage simply crashes. I have no errors in my error logs.
```
<?php
use \parallel\Runtime;
$runtime = new Runtime();

$task = static function (): string {
   for ($i = 0; $i < 20; $i++):
      echo " ";
   endfor;
   return "easy";
};

$future = $runtime->run($task);

for ($i = 0; $i < 20; $i++):
   echo "*";
endfor;

echo "<br />";
printf("\nUsing \\parallel\\Runtime is %s\n", $future->value());
die();
```
Any ideas what is going on here and how to approach it?

**Comment by dixyes** [comment]
fixed in 63c4fcd339e670556b152d76b8dcf3a34f268464
btw please report problems related to this pr in this pr thread

**Comment by falxen** [comment]
Looks like [63c4fcd](https://github.com/krakjoe/parallel/commit/63c4fcd339e670556b152d76b8dcf3a34f268464) fixed this issue. Thanks! Closing the issue.
